### PR TITLE
Fix #3333: Create BUILD.bazel file for domail/profile.

### LINKED ISSUE #3333: Create a BUILD.bazel in domain/profile
This task is part of the build system migration from Gradle to Bazel. 

## Prerequesites

- Make sure you are able to build Oppia using Bazel
- Follow the [Oppia Bazel Setup instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) to setup Bazel and learn the terminology.
- Read the [Gradle Bazel Migration Best Practies and FAQ](https://github.com/oppia/oppia-android/wiki/Gradle--Bazel-Migration-Best-Practices-and-FAQ) for more context and see if there's an answer there to any question you have.

## Task

1. Create a BUILD.bazel file in

`domain/src/main/java/org/oppia/android/domain/profile`

and its subdirectories.

1. Define the kt_android_library library/libraries

Based on the imports in the Kotlin files, add the required third_party dependencies to make the libraries build. Check the Useful links section for a list of third_party dependencies.  You may also need to add the non third_party dependencies.

3. Add the directory to the MIGRATED_PROD_FILES of the module's root BUILD.bazel file

4. Add the libraries to the list of deps dependencies in the targets that require the library

You will have to set the apropriate visbility. See the list of visibilities in the link below.

5. Clean up any unused deps dependencies

6. Verify that the app builds

## Useful links

- [BUILD.bazel files](https://github.com/oppia/oppia-android/search?q=filename%3ABUILD.bazel&type=code)
- [Visibilities](https://github.com/oppia/oppia-android/blob/76dc13dba3ad77aecd2218a5372949d6551f12fe/BUILD.bazel)
- [third_party dependencies](https://github.com/oppia/oppia-android/blob/a70d9b016f0d77dd30c4939041f82ea02ee28d1d/third_party/versions.bzl)

### PR DESCRIPTION
**Pull request by rishidyno**
Fix #3333 
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by rishidyno** [comment]
@BenHenning All tests have passes PTAL .

File: domain/src/main/java/org/oppia/android/domain/profile/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,29 @@
+"""
+Utilities for managing profile.
```
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/profile/BUILD.bazel]:
```suggestion
Domain definitions and controllers for managing profiles.
```

These are specific to the domain layer and aren't general-purpose utilities.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @rishidyno! Apologies for the late review. The PR looks good; just had 1 comment nit. PTAL.

**Comment by rishidyno** [comment]
@BenHenning PTAL changes were made

**Review by BenHenning — APPROVED** [review]
Thanks @rishidyno. LGTM!

**Pull request merged**